common/reconciliation/common/util.go

```diff
@@ -135,16 +149,34 @@ func Open(state int) bool {
 	return state == persistence.WorkflowStateCreated || state == persistence.WorkflowStateRunning
 }
 
+// Open returns true if execution state is open false if workflow is closed
+func ExecutionOpen(execution interface{}) bool {
+	return Open(getExecution(execution).State)
+}
+
+// getExecution returns base Execution
+func getExecution(execution interface{}) *Execution {
+	switch e := execution.(type) {
+	case *Execution:
```

Comment:
Seems here we are treating Execution to mean current? Cut concrete is called ConcreteExecution?

Reply:
Yes, you are right. the current type will be introduced as part of "introduce current execution type APIs" step